sdk/storage/storage-internal-avro/src/AvroParser.ts

```diff
@@ -28,32 +28,35 @@ export class AvroParser {
     return buf[0];
   }
 
+  // int and long are stored in variable-length zig-zag coding.
+  // variable-length: https://lucene.apache.org/core/3_5_0/fileformats.html#VInt
+  // zig-zag: https://developers.google.com/protocol-buffers/docs/encoding?csw=1#types
   private static async readZigZagLong(stream: AvroReadable): Promise<number> {
-    // copied from https://github.com/apache/avro/blob/master/lang/js/lib/utils.js#L321
-    let n = 0;
-    let k = 0;
-    let b, h, f, fk;
+    let zigZagEncoded = 0;
+    let significanceInBit = 0;
+    let byte, haveMoreByte, significanceInFloat;
 
     do {
-      b = await AvroParser.readByte(stream);
-      h = b & 0x80;
-      n |= (b & 0x7f) << k;
-      k += 7;
-    } while (h && k < 28);
-
-    if (h) {
-      // Switch to float arithmetic, otherwise we might overflow.
-      f = n;
-      fk = 268435456; // 2 ** 28.
+      byte = await AvroParser.readByte(stream);
+      haveMoreByte = byte & 0x80;
+      zigZagEncoded |= (byte & 0x7f) << significanceInBit;
+      significanceInBit += 7;
+    } while (haveMoreByte && significanceInBit < 28); // bitwise operation only works for 32-bit integers
+
+    if (haveMoreByte) {
+      // Switch to float arithmetic
+      // FIXME: this only works when zigZagEncoded is no more than Number.MAX_SAFE_INTEGER (2**53 - 1)
```

Comment:
please go through the FIXME section, and remove unncessary commented out codes. If it's really a TODO not iniclude in MVP, please directly open github issues to track.